### PR TITLE
Conditional set output is wrong

### LINKED ISSUE #5918: Conditional Set "output" is wrong
#### Summary
When no data is present for telemetry in use by a Condition Set (CS), the set _says_ it's outputting a default value (orange arrow), but it actually is not. Note that the Default condition is not hilited (green arrow).

Screen Shot 2022-10-26 at 4 09 33 PM

LAD Table view of the Condition set shows that no data is available:

Screen Shot 2022-10-26 at 4 18 50 PM

Adding the CS to a Display Layout as an alpha shows that no value is being output:

Screen Shot 2022-10-26 at 4 19 56 PM

#### Expected vs Current Behavior
The CS should not erroneously report an output value, but instead should be blank or "---". CS views currently only output a value when they have an input, such as data. This makes sense: the CS needs an input to determine its output frequency. 

This behavior has side-effects on how Conditional Styling works. Also, the pivoted LAD view of the CS itself is counter to how that control works in all other views: that view shows the output of the compositional _children_ of the CS, not the output of the CS itself. @charlesh88 to file additional bugs to fix this.

#### Steps to Reproduce

1. Make or edit a CS and include one or more telemetry points that don't have any data currently. Give it at least one condition besides the default.
2. Save the CS, then nav away and back.
3. Observe: the CS reports an output of Default, but shouldn't.

#### Environment

* Open MCT Version:  2.0.5, 2.1.2
* Deployment Type:  MMOC, /latest
* OS: any
* Browser: any

#### Impact Check List

- [x] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by scottbell**
Closes #5918 

### Describe your changes:

- Removed default output from being used before component has received and evaluated valid telemetry.
- Added check in staleness for existence of staleness response (to prevent thrown error)

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by scottbell** [comment]
Here's the new behavior:

https://user-images.githubusercontent.com/9853862/216042371-b551f935-2dab-4a7d-809b-bfee4e6f4357.mov

I'm turning of YAMCS data to simulate the "no data" situation, then starting YAMCS to show the ConditionSet receiving data. If this looks good, I'll add an e2e test in the OpenMCT YAMCS project.

**Comment by unlikelyzero** [comment]
https://www.w3.org/WAI/ARIA/apg/patterns/accordion/ is a good guideline for the rules within the condition set.

Mainly isExpanded and button

**Comment by shefalijoshi** [comment]
@scottbell @charlesh88  If you drag the condition set with output `---` into a display layout, it displays a different output. Expectation is that it should also show `---` right?

https://user-images.githubusercontent.com/5247066/217098392-761647e8-6ee8-476a-90b2-cb28cca230c7.mov

**Comment by scottbell** [comment]
> 

Be sure the Sine Wave Generators you're testing with have a somewhat low frequency and substantial delay. Also, changing the delay of a Sine Wave Generator necessitates a page reload (from what I can tell):

https://user-images.githubusercontent.com/9853862/217101296-4d44cfd0-fcc6-44f8-bc8c-1ff7b9e3c5d2.mov

**Comment by scottbell** [comment]
@shefalijoshi @charlesh88 Regarding the difference between DisplayLayouts and the ConditionSet view on historical data, this appears to be a pre-existing issue in `master` from what I can tell:

https://user-images.githubusercontent.com/9853862/217257250-49c889a5-78b6-4080-ad96-4a80e78366ed.mov

Namely, the ConditionSet view only updates if it's Local Clock mode:

https://user-images.githubusercontent.com/9853862/217257506-6d7b444d-9826-4a17-a988-ab89f492f13a.mov

**Comment by akhenry** [comment]
> @scottbell @charlesh88 If you drag the condition set with output `---` into a display layout, it displays a different output. Expectation is that it should also show `---` right?
> ConditionSetDefaultOutput.mov

Good catch. I think the issue here is that a condition set that is dragged into a display layout is just treated like any other telemetry point. When there is no data we show nothing right now. It is admittedly inconsistent with the Condition Set view, but we're at least not showing _incorrect_ state any more.

We could discuss what Display Layouts show when a telemetry point has no value in bounds, but I'm happy to treat that as a separate issue for now.

**Review by charlesh88 — COMMENTED** [review]
See no issues with this implementation, but I'm also not reviewing this from a coding perspective at all - this is only a comment.

**Comment by charlesh88** [comment]
@scottbell Test data in Condition Sets should not be affected at all by Time Conductor settings. If you're seeing issues in that regard, that's a problem. I just tested this in a Condition Set in 2.1.4 in the MMOC and saw no problems.

**Comment by scottbell** [comment]
> @scottbell Test data in Condition Sets should not be affected at all by Time Conductor settings. If you're seeing issues in that regard, that's a problem. I just tested this in a Condition Set in 2.1.4 in the MMOC and saw no problems.

It’s not. Test data behaves the same in this branch as it does in `master`.

File: e2e/tests/functional/plugins/conditionSet/conditionSet.e2e.spec.js [hunk 1]
```diff
@@ -199,4 +199,81 @@ test.describe('Basic Condition Set Use', () => {
         const numOfUnnamedConditions = await page.locator('text=Unnamed Condition').count();
         expect(numOfUnnamedConditions).toEqual(1);
     });
+    test('ConditionSet should ouput blank instead of the default value', async ({ page }) => {
```
Comment (360macky) [inline e2e/tests/functional/plugins/conditionSet/conditionSet.e2e.spec.js:202]: NIT: _output_ misspelled.

**Comment by scottbell** [comment]
@shefalijoshi link to [conversation summation](https://github.com/nasa/openmct/issues/5918#issuecomment-1419809006).

**Review by charlesh88 — CHANGES_REQUESTED** [review]
1. The view isn't updating as expected when saving and exiting. Specifically: edit a CS, add telemetry (in this case a Sine Wave gen) and see that the Default condition is matched. Remove the telemetry, but still see Default being output. Save and exit - and still, output is the Default. Nav away and back, and now I see "---" as expected. It should ideally display "---" as soon as no input is available for a matching evaluation.
1. The issue needs testing notes - I added based on the original Steps to Reproduce. Can you have a look and augment as needed?
1. I expected to see "---" in the output of the CS when added to a Display Layout as an alphanumeric:
Screen Shot 2023-03-06 at 1 14 30 PM
The CS does display "---" in a LAD Table view, and it'd be nice for it to be consistent. However, that is not a common use case and I don't feel the need to press for this.

**Comment by scottbell** [comment]
> 1. The view isn't updating as expected when saving and exiting. Specifically: edit a CS, add telemetry (in this case a Sine Wave gen) and see that the Default condition is matched. Remove the telemetry, but still see Default being output. Save and exit - and still, output is the Default. Nav away and back, and now I see "---" as expected. It should ideally display "---" as soon as no input is available for a matching evaluation.
> 
>     2. The issue needs testing notes - I added based on the original Steps to Reproduce. Can you have a look and augment as needed?
> 
>     3. I expected to see "---" in the output of the CS when added to a Display Layout as an alphanumeric. The CS does display "---" in a LAD Table view, and it'd be nice for it to be consistent. However, that is not a common use case and I don't feel the need to press for this.

Appreciate the review, @charlesh88 - I'll take a look!

**Comment by scottbell** [comment]
Added some fixes for deleting telemetry and immediately showing `---`

https://user-images.githubusercontent.com/9853862/223426094-247096e9-a65b-49a5-ab7f-318c44a8177d.mov

**Comment by akhenry** [comment]
In testing this I found a weird edge-case, which I will file a followup for because I think it's actually a long-standing issue with conditional styling and not an issue with condition sets themselves.

1. Drag a condition set with no flowing telemetry (as created here) into a display layout
2. Define some conditional style rules
3. Save
4. Note that whichever conditional style was _last selected_ remains active in the display layout. Navigating away clears it.

File: src/plugins/condition/components/ConditionCollection.vue [hunk 2]
```diff
@@ -158,26 +154,23 @@ export default {
...
-        this.updateDefaultCondition();
+        this.conditionManager.on('noTelemetryObjects', this.handleNoTelemetryObjects);
         this.stalenessSubscription = {};
     },
     methods: {
         handleConditionSetResultUpdated(data) {
             this.currentConditionId = data.conditionId;
             this.$emit('conditionSetResultUpdated', data);
         },
+        handleNoTelemetryObjects(data) {
```
Comment (akhenry) [inline src/plugins/condition/components/ConditionCollection.vue:165]: handler functions should be named for what they do. In this case `emitNoTelemetryObjectEvent` or something.

**Comment by akhenry** [comment]
Filed https://github.com/nasa/openmct/issues/6444 for the followup issue

**Review by akhenry — APPROVED** [review]

**Review by charlesh88 — CHANGES_REQUESTED** [review]
I'm now seeing composition problems. Specifically:
1. Make a new Condition Set (CS).
2. Add telem to it (sine wave gen). Define condition(s).
3. Save and exit editing.
4. In the tree, expand the CS.
5. Use the "Remove" action to remove the telem element. The element will appear to be removed. The CS outputs "---"  but incorrectly highlights the Default condition.
6. Collapse the CS in the tree, then re-expand it. Note that the telem element returns, and is clickable/navigable. The CS is not evaluating, and displays "---".
7. Edit the CS, note that the telem element appears in the CS's Elements pool and the CS does not evaluate it.

I tested this with another domain object (Gauge) and did not see the same issue. No console errors were thrown during the above sequence.

Default condition hilited after telem element removed
Screen Shot 2023-03-16 at 9 23 44 AM

Telem element reappears after toggle in tree
Screen Shot 2023-03-16 at 9 24 02 AM

**Comment by scottbell** [comment]
@charlesh88 I think the composition issue is separate from this PR, in that the behavior is also happening in `master`. It may be due to this [recent PR](https://github.com/nasa/openmct/pull/6437)? Here's a screenshare of it on `master`:


https://user-images.githubusercontent.com/9853862/225709277-906c3ac2-1a09-47a8-99c1-2077b5cd88d9.mov

**Comment by scottbell** [comment]
> @charlesh88 I think the composition issue is separate from this PR, in that the behavior is also happening in `master`. It may be due to this [recent PR](https://github.com/nasa/openmct/pull/6437)? Here's a screenshare of it on `master`:

Tested before that other recent PR in, and it looks like this bug predates that too.

**Comment by scottbell** [comment]
> > @charlesh88 I think the composition issue is separate from this PR, in that the behavior is also happening in `master`. It may be due to this [recent PR](https://github.com/nasa/openmct/pull/6437)? Here's a screenshare of it on `master`:
> 
> Tested before that other recent PR in, and it looks like this bug predates that too.

@charlesh88 talked to @akhenry about this, and we're going to file a separate issue for the tree composition problem. Regarding this:
>The CS outputs "---" but incorrectly highlights the Default condition.
I'll look into it for this PR.

**Pull request merged**